### DISCUSSION TITLE
Bind a resolver to a class

### DISCUSSION THREAD

**Discussion by TerranWorks** [mirumee/ariadne#966] [Q&A]
In the documentation it shows that a class can be used as a resolver but I couldn't find an example of how to bind it. I'd like to use the decorator method but I can't figure out where to put it. I'd like to use an sqlmodel as the resolver itself so this would be really useful information.

**Comment by rafalp** [comment]
No way for using decorator method with class comes to mind here, unless you want to decorate `@classmethod` or `@staticmethod` methods. Otherwise you'll have to use `user_type.set_field("field", user_class_instance.resolve_field)`.

What are you trying to accomplish? Maybe [Ariadne GraphQL Modules](https://github.com/mirumee/ariadne-graphql-modules) will be a better fit for your issue?

**Reply by TerranWorks** [reply]
Ariadne GraphQL Modules looks interesting, Ill have to look into that . Doing a bit more research I just misunderstood how the default resolver works. The problem I'm having now is that the methods in my class are being called by the default resolver but they're complaining that they only have one parameter but two were given. I'm not sure what parameters the resolver is trying to pass to my methods. Here is an example method inside my class and a resolver which returns that type:

```python
class BlogPost(Base):
    """Sqlalchemy model for BlogPosts."""

    def owner(self):
        """Gets the owner of this blog post."""
        session = object_session(self)
        return session.query(core.Profile).get(self.owner_id)

@mutation.field("createBlogPost")
def resolve_create_blog_post(_, info, title, username, markdown, html):
    try:
        session = info.context["db_session"]
        owner = core.get_profile_by_username(session, username)
        blog_post = blog.create_blog_post(
            session, owner=owner, title=title, markdown=markdown, html=html
        )
        payload = {"success": True, "blogPost": blog_post}
    except CustomException as e:
        payload = {"success": False, "errors": [repr(e)]}

    return payload
```
and the schema

```graphql
type Mutation {
    createBlogPost(title:String!, username:String!, markdown:String!, html:String!): BlogPostPayload!
}
type BlogPost {
    title: String
    owner: Profile
    content_meta: Content
    markdown: String
    html: String
}
type BlogPostPayload {
    blogPost: BlogPost
    success: Boolean
    errors: [String]
}
```
